cpp/include/rolling.hpp

```diff
@@ -37,6 +41,10 @@ namespace cudf {
  *   to indicate which elements are valid.
  * - support for dynamic rolling windows, i.e. window size or number of
  *   observations can be specified for each element using an additional array.
+ * This function is asynchronous with respect to the CPU, i.e. the call will
```

Comment:
Updated the description and removed the stream argument. Note that the function is still async with respect to the CPU since I don't have device sync explicitly in the implementation. I believe all our other functions behave like that - i.e. I looked into `gather` and it's the same. The only scenario when they will be executed synchronously is when the library is compiled in a non-release mode (I use `CHECK_STREAM`)